IGAPP-1128: Replaced deprecated keyCode event property

Cmd190 opened the pull request:
KeyboardEvent.keyCode was replaced.

This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

LeandraH reviewed (approved):
Nice!

SteffiStuffel commented on web/src/hooks/useOnClickOutside.ts:
```diff
@@ -12,11 +12,11 @@ const useOnClickOutside = (ref: { current: HTMLElement | null }, callback: () =>
-      const KEY_ENTER = 13
-      const KEY_ESCAPE = 27
-      if (event.keyCode === KEY_ESCAPE) {
+      const KEY_ENTER = 'Enter'
```
I think now these constants are not needed any longer, as the naming is already clear.

Cmd190 replied:
done

steffenkleinle reviewed (commented):
Tested on firefox. Minor since it probably only affects a rare minority of users like me: I switched Escape and Caps Lock on my keyboard but I still have to use the same keys here, i.e. `Escape` to close instead of `Caps Lock`. Before it worked fine for me to close with `Caps Lock`. Maybe you could investigate if there is a way to fix this?

Cmd190 commented:
> Tested on firefox. Minor since it probably only affects a rare minority of users like me: I switched Escape and Caps Lock on my keyboard but I still have to use the same keys here, i.e. `Escape` to close instead of `Caps Lock`. Before it worked fine for me to close with `Caps Lock`. Maybe you could investigate if there is a way to fix this?

Yes this was because it checked for the code send from the keyboard and not the actual value. I changed to the `key` event which should do the jobt

steffenkleinle reviewed (approved):
Good stuff!
